### PR TITLE
cmake: cleanups

### PR DESCRIPTION
**Pull request by tchaikov**
src/CMakeLists.txt is sort of cluttered. so let's clean it up a little bit.

### CODE REVIEW

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -628,6 +581,9 @@ endif()
 
 add_library(common-objs OBJECT ${libcommon_files})
 set(ceph_common_objs
+  $<TARGET_OBJECTS:common-msg>
+  $<TARGET_OBJECTS:common-msg-async>
+  $<TARGET_OBJECTS:common-msg-simple>
```
Comment (tchaikov) [inline src/CMakeLists.txt]: it's difficult to build the components like msg, auth src/common as static libraries as we have inter-dependencies between them.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-cmake-cleanups-kefu/b254095801cc35786bd1f1a1a5499f82ae8dce11/

**Comment by smithfarm** [comment]
@tchaikov Given that ceph-dencoder is shipped as a part of the ceph-common package, why are we now building it in `src/test`?

Since CMake only descends into `src/tests` if WITH_TESTS is true, this PR breaks the RPM build when building with `-DWITH_TESTS=OFF`:

```
[ 6599s] error: File not found: /home/abuild/rpmbuild/BUILDROOT/ceph-14.0.0-lp150.1510.1.x86_64/usr/bin/ceph-dencoder
[ 6599s] 
[ 6599s] 
[ 6599s] RPM build errors:
[ 6599s]     File not found: /home/abuild/rpmbuild/BUILDROOT/ceph-14.0.0-lp150.1510.1.x86_64/usr/bin/ceph-dencoder
[ 6599s] 
```

**Comment by tchaikov** [comment]
@smithfarm i think it's an oversight. will move it to `src/tools`.

File: cmake/modules/Finddpdk.cmake [hunk 2]
```diff
@@ -19,10 +19,16 @@ set(components
...
-
-foreach(c ${components})
   list(APPEND check_LIBRARIES "${DPDK_rte_${c}_LIBRARY}")
+  if(DPDK_rte_${c}_LIBRARY)
+    add_library(dpdk::${c} UNKNOWN IMPORTED)
+    set_target_properties(dpdk::${c} PROPERTIES
+      INTERFACE_INCLUDE_DIRECTORIES "${DPDK_INCLUDE_DIR}"
+      IMPORTED_LOCATION "${DPDK_rte_${c}_LIBRARY}")
+    target_compile_options(dpdk::${c} INTERFACE
+      "-march=native")
```
Comment (smithfarm) [inline cmake/modules/Finddpdk.cmake]: Should this have been `-march=core2` in light of http://tracker.ceph.com/issues/24948 ?

Comment (tchaikov) [reply cmake/modules/Finddpdk.cmake]: as the title states, it's a cleanup. so i was not trying to change the behavior here.

File: src/CMakeLists.txt [hunk 3]
```diff
@@ -692,15 +691,9 @@ if(WITH_DPDK)
...
     ${common_async_dpdk_files})
-  # to ready the dpdk header files
-  if(TARGET dpdk-ext)
-    add_dependencies(common_async_dpdk dpdk-ext)
-  endif()
   list(APPEND ceph_common_deps common_async_dpdk)
   target_link_libraries(common_async_dpdk
     ${DPDK_LIBRARIES})
-  set_target_properties(common_async_dpdk PROPERTIES
-    COMPILE_FLAGS "-march=native -I${DPDK_INCLUDE_DIR}")
```
Comment (smithfarm) [inline src/CMakeLists.txt]: ah, coming from here - got it

**Pull request merged**